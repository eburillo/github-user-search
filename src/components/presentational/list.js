import React, { Component } from 'react';
import RepoContainer from '../containers/repo-container.js';

export class List extends Component {

	render() {
		return (
			<ul className="repo-list">
				{this.props.repos.map(this.displayRepo)}
			</ul>
		);
	}

	displayRepo(repo, i) {
		return (
			<RepoContainer key={i} repo={repo}></RepoContainer>
		);
	}
}

export default List;
