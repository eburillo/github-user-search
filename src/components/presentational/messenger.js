import React, { Component } from 'react';

export class Messenger extends Component {

	render() {
		return (
			<p>
				{this.props.message}
			</p>
		);
	}

}

export default Messenger;
