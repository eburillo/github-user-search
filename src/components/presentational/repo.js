import React, { Component } from 'react';

export class Repo extends Component {

	render() {
		return (
			<li className="repo-item">
				<a className="repo-link" target="_blank" href={this.props.url}>{this.props.name}</a>
				<p className="repo-desc">{this.props.desc}</p>
			</li>
		);
	}
}

export default Repo;
