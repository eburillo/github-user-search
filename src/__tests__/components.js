import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {List} from '../components/presentational/list.js';
import {Messenger} from '../components/presentational/messenger.js';
import {Repo} from '../components/presentational/repo.js';

describe('Render Presentational components', function() {
  it('renders a list in List component', () => {
    const list = [
          {
            "name": "atomic-design",
            "description": "The repository for the Atomic Design book",
            "html_url": "https://github.com/bradfrost/atomic-design"
          }
        ];

    let tree = renderer.create(
      <List repos={list}>
      </List>
    );
    expect(tree).toMatchSnapshot();
  });

  it('renders a message in Messeger component', () => {
    let message = "This is a testing message.";
    let tree = renderer.create(
      <Messenger message={message}>
      </Messenger>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders an item in Repo component', () => {
    let repo = {
      "name": "atomic-design",
      "description": "The repository for the Atomic Design book",
      "html_url": "https://github.com/bradfrost/atomic-design"
    };
    let tree = renderer.create(
      <Repo name={repo.name} url={repo.html_url} desc={repo.description}>
      </Repo>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
