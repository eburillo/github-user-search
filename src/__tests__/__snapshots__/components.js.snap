exports[`Render Presentational components renders a list in List component 1`] = `
<ul
  className="repo-list">
  <li
    className="repo-item">
    <a
      className="repo-link"
      href="https://github.com/bradfrost/atomic-design"
      target="_blank">
      atomic-design
    </a>
    <p
      className="repo-desc">
      The repository for the Atomic Design book
    </p>
  </li>
</ul>
`;

exports[`Render Presentational components renders a message in Messeger component 1`] = `
<p>
  This is a testing message.
</p>
`;

exports[`Render Presentational components renders an item in Repo component 1`] = `
<li
  className="repo-item">
  <a
    className="repo-link"
    href="https://github.com/bradfrost/atomic-design"
    target="_blank">
    atomic-design
  </a>
  <p
    className="repo-desc">
    The repository for the Atomic Design book
  </p>
</li>
`;
